### PR TITLE
Do not use the listener names from Kafka in docs, use the names from CRs

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix
- Documentation

### Description

As described in #2028, the docs should not be using the listener names used in the Kafka configuration (e.g. EXTERNAL, CLIENT, CLIENTTLS or REPLICATION) but rather use the names we use in the KAfka custom resources (plain, tls, external).

### CODE REVIEW

File: documentation/book/con-tls-connections.adoc [hunk 1]
```diff
@@ -26,8 +26,8 @@ Like the Cluster Operator, the Topic and User Operators each use a TLS sidecar w
 
 == Kafka Client connections
 
-Encrypted communication between Kafka brokers and clients running within the same Kubernetes cluster is provided through the `CLIENTTLS` listener on port 9093.
+Encrypted communication between Kafka brokers and clients running within the same Kubernetes cluster is provided through the `tls` listener on port 9093.
 
-Encrypted communication between Kafka brokers and clients running outside the same Kubernetes cluster is provided through the `EXTERNAL` listener on port 9094.
+Encrypted communication between Kafka brokers and clients running outside the same Kubernetes cluster is provided through the `external` listener (the port of the external listener depends on its type).
```
Comment (ppatierno) [inline documentation/book/con-tls-connections.adoc]: `external` the second time as well?

File: documentation/book/con-tls-connections.adoc [hunk 2]
```diff
@@ -16,7 +16,7 @@ This TLS encrypting `stunnel` proxy is instantiated from the `spec.zookeeper.stu
 
 == Kafka interbroker communication
 
-Communication between Kafka brokers is done through the `REPLICATION` listener on port 9091, which is encrypted by default.
+Communication between Kafka brokers is done through an internal listener on port 9091, which is encrypted by default and not accessible to {ProductName} users.
```
Comment (tombentley) [inline documentation/book/con-tls-connections.adoc]: I think "not accessible to Kafka clients" would be slightly clearer.

File: documentation/book/con-tls-connections.adoc [hunk 3]
```diff
@@ -26,8 +26,8 @@ Like the Cluster Operator, the Topic and User Operators each use a TLS sidecar w
 
 == Kafka Client connections
 
-Encrypted communication between Kafka brokers and clients running within the same Kubernetes cluster is provided through the `CLIENTTLS` listener on port 9093.
+Encrypted communication between Kafka brokers and clients running within the same Kubernetes cluster is provided through the `tls` listener on port 9093.
```
Comment (tombentley) [inline documentation/book/con-tls-connections.adoc]: Something like "...can be provided by configuring the `spec.kafka.listeners.tls` listener, which listens on port 9093" is a bit more explicit about what exactly what `tls` we're referring to and that isn't not configured by default.

File: documentation/book/con-tls-connections.adoc [hunk 4]
```diff
@@ -26,8 +26,8 @@ Like the Cluster Operator, the Topic and User Operators each use a TLS sidecar w
...
 
-Encrypted communication between Kafka brokers and clients running outside the same Kubernetes cluster is provided through the `EXTERNAL` listener on port 9094.
+Encrypted communication between Kafka brokers and clients running outside the same Kubernetes cluster is provided through the `external` listener (the port of the `external` listener depends on its type).
```
Comment (tombentley) [inline documentation/book/con-tls-connections.adoc]: Similar rewording here?

File: documentation/book/con-tls-connections.adoc [hunk 5]
```diff
@@ -26,8 +26,8 @@ Like the Cluster Operator, the Topic and User Operators each use a TLS sidecar w
...
 
-NOTE: You can use the `CLIENT` listener on port 9092 for unencrypted communication with brokers.
+NOTE: You can use the `plain` listener on port 9092 for unencrypted communication with brokers.
```
Comment (tombentley) [inline documentation/book/con-tls-connections.adoc]: NOTE: Unencrypted client communication with brokers is configured by `spec.kafka.listeners.plain`, which listens on port 9092.

**Pull request merged**